TOY-39 MainActivity 제거

pby2017 opened the pull request:
1. MainActivity 클래스를 제거했습니다.

2. CameraActivity 코드를 CameraFragment로 옮겼습니다.

3. CameraActivity에 대해 권한 설정 조작 등 OS에 의해 종료되면 onCreate()부터 다시 시작하도록 수정했습니다.
: onSaveInstanceState() 재정의

4. CameraViewModel의 context를 global application 의 context로 대체했습니다.

5. xml의 tools:context가 MainActivity인 코드를 제거했습니다.

6. strings에서 MainActivity를 CameraFragment로 수정했습니다.

pby2017 committed:
TOY-39 MainActivity 제거

1. MainActivity 클래스를 제거했습니다.

2. CameraActivity 코드를 CameraFragment로 옮겼습니다.

3. CameraActivity에 대해 권한 설정 조작 등 OS에 의해 종료되면 onCreate()부터 다시 시작하도록 수정했습니다.

4. CameraViewModel의 context를 global application 의 context로 대체했습니다.

5. xml의 tools:context가 MainActivity인 코드를 제거했습니다.

6. strings에서 MainActivity를 CameraFragment로 수정했습니다.

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/ui/camera/capture/CameraViewModel.kt:
```diff
@@ -14,25 +13,25 @@ import com.otaliastudios.cameraview.Grid
-    val gridButtonText: ObservableField<String> = ObservableField(context.getString(R.string.text_grid))
+class CameraViewModel : ViewModel() {
+
+    val showMoreButtonSrc: ObservableField<Drawable?> = ObservableField(ContextCompat.getDrawable(GlobalApplication.globalApplicationContext, R.drawable.baseline_more_horiz_black_24))
```
GlobalApplication.globalApplicationContext 이름이 너무 길어서 차라리 변수로 빼서 쓰는게 가독성에 더 좋을 거 같네요

pby2017 replied:
네,
피드백 반영해서 아래 사항과 같이 수정하겠습니다.

1. `GlobalApplication.globalApplicationContext` 변수명을 `appContext`로 수정
2. context 없이 resource int 값을 사용해도 되는 경우 수정

bonchangkoo reviewed (commented):
코멘트 확인 부탁해용

pby2017 committed:
TOY-39 피드백 반영

1. GlobalApplication.globalApplicationContext 변수명을 appContext로 수정
2. context 사용하지 않아도 되는 경우 제거

pby2017 commented:
> 코멘트 확인 부탁해용

@bonchangkoo 님,
피드백 반영해서 코드 수정했습니다.

코드 확인 부탁드립니다.

bonchangkoo reviewed (approved):
확인했습니다.
